### PR TITLE
feature/serene-744

### PR DESCRIPTION
**Pull request by gaj67**
Test EpgManagementController, especially that with 2 consecutive creates, the 1st works and the 2nd fails (status 400); similarly for 2 consecutive deletes.

Also, check to see if http://localhost:8080/api/v1.0/indexes complains that it cannot find the page - this is weird, but actually okay for a machine-only API!

### CODE REVIEW

File: search/src/test/java/au/com/d2dcrc/ia/search/management/EpgManagementControllerIT.java [hunk 1]
```diff
@@ -149,6 +155,19 @@ private EpgMetaView createEpg(
...
+        final String path = "/api/v1.0/indexes/" + name;
+
+        given()
+            .contentType(ContentType.JSON).body(epgReferenceModel)
+
+            .when().post(path)
+
+            .then()
+            .contentType(ContentType.JSON)
+            .statusCode(HttpStatus.BAD_REQUEST.value());
```
Comment (tmoschou) [inline search/src/test/java/au/com/d2dcrc/ia/search/management/EpgManagementControllerIT.java]: Might be worth asserting the response body against the error-view.schema.json (which is currently outdated)
`.body(matchesJsonSchema(getClass().getResource("error-view.schema.json")))`

File: search/src/test/java/au/com/d2dcrc/ia/search/management/EpgManagementControllerIT.java [hunk 2]
```diff
@@ -192,4 +211,15 @@ private void deleteEpg(final String name) {
...
 
+    private void deleteEpgWhenNotExists(final String name) {
+        given()
+            .contentType(ContentType.JSON)
+
+            .when()
+            .delete("/api/v1.0/indexes/" + name)
+
+            .then()
+            .statusCode(HttpStatus.BAD_REQUEST.value());
```
Comment (tmoschou) [inline search/src/test/java/au/com/d2dcrc/ia/search/management/EpgManagementControllerIT.java]: same here

File: search/src/main/java/au/com/d2dcrc/ia/search/common/RequestInfo.java [hunk 3]
```diff
@@ -38,4 +41,44 @@ public static String getRemoteIpAddress(final HttpServletRequest servletRequest)
...
+     * 
+     * @param request - The HTTP request.
+     * @return The HTTP status code.
+     */
+    public static HttpStatus getStatus(HttpServletRequest request) {
+        Integer statusCode = (Integer) request.getAttribute("javax.servlet.error.status_code");
+        if (statusCode != null) {
+            try {
+                return HttpStatus.valueOf(statusCode);
+            } catch (Exception ex) {
```
[the remark this answers is not in the document]
Comment (gaj67) [reply search/src/main/java/au/com/d2dcrc/ia/search/common/RequestInfo.java]: Done

File: search/src/main/java/au/com/d2dcrc/ia/search/response/ErrorControllerConfig.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ * Configures the Whitelabel Error Page.
+ */
+@Configuration
+public class ErrorControllerConfig {
+
+    @Autowired
+    private ErrorAttributes errorAttributes;
+
+    @Bean
+    public AppErrorController appErrorController() {
```
[the remark this answers is not in the document]
Comment (gaj67) [reply search/src/main/java/au/com/d2dcrc/ia/search/response/ErrorControllerConfig.java]: Deleted

File: search/src/test/java/au/com/d2dcrc/ia/search/management/EpgManagementControllerIT.java (hunk 1 above)
Comment (gaj67) [reply search/src/test/java/au/com/d2dcrc/ia/search/management/EpgManagementControllerIT.java]: Done

File: search/src/test/java/au/com/d2dcrc/ia/search/management/EpgManagementControllerIT.java (hunk 2 above)
Comment (gaj67) [reply search/src/test/java/au/com/d2dcrc/ia/search/management/EpgManagementControllerIT.java]: Done

**Pull request merged**